### PR TITLE
test: Clean up Flake warnings on assert_(not_)?called

### PR DESCRIPTION
**Pull request by RyanSkonnord**

### CODE REVIEW

**Review by wedamija — APPROVED** [review]

File: tests/sentry/api/serializers/test_project.py [hunk 1]
```diff
@@ -473,7 +473,7 @@ def test_stats_with_sessions(
         assert results[0]["sessionStats"]["currentCrashFreeRate"] == 75.63453
         assert results[0]["sessionStats"]["hasHealthData"]
 
-        check_has_health_data.assert_not_called()
```
Comment (asottile-sentry) [inline tests/sentry/api/serializers/test_project.py:476]: these functions are fine and work correctly in python 3.8 -- I suspect you have an ancient `flake8-sentry` installed which we deleted a few months ago

**Review by asottile-sentry — CHANGES_REQUESTED** [review]

File: tests/sentry/api/serializers/test_project.py (hunk 1 above)
Comment (RyanSkonnord) [reply tests/sentry/api/serializers/test_project.py:476]: How would I go about fixing that? I tried all relevant `make` commands I could find and I'm still getting

```
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

tests/sentry/api/serializers/test_project.py:476:9: S001: Avoid using the assert_not_called mock call as it is confusing and prone to causing invalid test behavior.
tests/sentry/api/serializers/test_project.py:505:9: S001: Avoid using the assert_called mock call as it is confusing and prone to causing invalid test behavior.
tests/sentry/tasks/test_post_process.py:83:9: S001: Avoid using the assert_not_called mock call as it is confusing and prone to causing invalid test behavior.
tests/sentry/tasks/test_post_process.py:84:9: S001: Avoid using the assert_not_called mock call as it is confusing and prone to causing invalid test behavior.
tests/sentry/tasks/test_post_process.py:85:9: S001: Avoid using the assert_not_called mock call as it is confusing and prone to causing invalid test behavior.
tests/sentry/tasks/test_post_process.py:88:9: S001: Avoid using the assert_not_called mock call as it is confusing and prone to causing invalid test behavior.
```

Comment (asottile-sentry) [reply tests/sentry/api/serializers/test_project.py:476]: `pip uninstall flake8-sentry`

Comment (RyanSkonnord) [reply tests/sentry/api/serializers/test_project.py:476]: I get

```
$ pip uninstall flake8-sentry
WARNING: Skipping flake8-sentry as it is not installed.
```

(Both in and out of direnv.)

Comment (asottile-sentry) [reply tests/sentry/api/serializers/test_project.py:476]: it might be called sentry-flake8

Comment (RyanSkonnord) [reply tests/sentry/api/serializers/test_project.py:476]: Got it, thanks.

**Comment by RyanSkonnord** [comment]
Warning was out of date.

**Pull request closed without merging**